modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryServiceImpl.java

```diff
@@ -450,10 +457,48 @@ private void _checkPortletResourcePermission(
 		ObjectDefinition objectDefinition =
 			_objectDefinitionPersistence.findByPrimaryKey(objectDefinitionId);
 
-		if (!objectDefinition.isAccountEntryRestricted()) {
+		if (!objectDefinition.isRootDescendantNode() &&
+			!objectDefinition.isAccountEntryRestricted()) {
+
 			return;
 		}
 
+		if (objectDefinition.isRootDescendantNode()) {
+			ObjectDefinition rootObjectDefinition =
+				_objectDefinitionPersistence.findByPrimaryKey(
+					objectDefinition.getRootObjectDefinitionId());
+
+			if (!rootObjectDefinition.isAccountEntryRestricted()) {
+				return;
+			}
+
+			TreeFactory treeFactory = _treeFactorySnapshot.get();
+
+			Tree tree = treeFactory.create(
+				objectDefinition.getRootObjectDefinitionId());
+
+			Node node = tree.getNode(objectDefinition.getObjectDefinitionId());
+
+			Edge edge = node.getEdge();
+
+			ObjectRelationship objectRelationship =
+				_objectRelationshipLocalService.getObjectRelationship(
+					edge.getObjectRelationshipId());
+
+			ObjectField objectField = _objectFieldLocalService.getObjectField(
+				objectRelationship.getObjectFieldId2());
+
+			ObjectEntry objectEntry = objectEntryLocalService.getObjectEntry(
+				MapUtil.getLong(values, objectField.getName()));
+
+			objectEntry = objectEntryLocalService.getObjectEntry(
+				objectEntry.getRootObjectEntryId());
+
+			values = objectEntry.getValues();
+
+			objectDefinition = rootObjectDefinition;
+		}
```

Comment:
@p-albuquerque can we use the rootObjectEntryId stored in ObjectEntryTable to get these values?

Reply:
This objectEntry have no rootObjectEntryId yet (just in objectEntryLocalService this column will be filled). So I thought in get his direct parent and take the rootObjectEntry from him.